src/Features/Core/Portable/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer.InProcOrRemoteHostAnalyzerRunner.cs

```diff
@@ -12,145 +12,219 @@
 
-                return DiagnosticAnalysisResultMap.Create(
-                    builderMap.ToImmutableDictionary(kv => kv.Key, kv => DiagnosticAnalysisResult.CreateFromBuilder(kv.Value)),
-                    analysisResult.AnalyzerTelemetryInfo);
-            }
+            var result = builderMap.ToImmutableDictionary(kv => kv.Key, kv => DiagnosticAnalysisResult.CreateFromBuilder(kv.Value));
+            var telemetry = getTelemetryInfo
+                ? analysisResult.AnalyzerTelemetryInfo
+                : ImmutableDictionary<DiagnosticAnalyzer, AnalyzerTelemetryInfo>.Empty;
+            return DiagnosticAnalysisResultMap.Create(result, telemetry);
+        }
 
-            private async Task FireAndForgetReportAnalyzerPerformanceAsync(Project project, RemoteHostClient? client, AnalysisResult analysisResult, CancellationToken cancellationToken)
+        private async Task FireAndForgetReportAnalyzerPerformanceAsync(
+            DocumentAnalysisScope? documentAnalysisScope,
+            Project project,
+            RemoteHostClient? client,
+            AnalysisResult analysisResult,
+            CancellationToken cancellationToken)
+        {
+            if (client == null)
             {
-                if (client == null)
-                {
-                    return;
-                }
+                return;
+            }
 
-                try
-                {
-                    await client.RunRemoteAsync(
-                        WellKnownServiceHubService.CodeAnalysis,
-                        nameof(IRemoteDiagnosticAnalyzerService.ReportAnalyzerPerformance),
-                        solution: null,
-                        new object[]
-                        {
-                            analysisResult.AnalyzerTelemetryInfo.ToAnalyzerPerformanceInfo(_analyzerInfoCache),
+            try
+            {
+                await client.RunRemoteAsync(
+                    WellKnownServiceHubService.CodeAnalysis,
+                    nameof(IRemoteDiagnosticAnalyzerService.ReportAnalyzerPerformance),
+                    solution: null,
+                    new object[]
+                    {
+                            analysisResult.AnalyzerTelemetryInfo.ToAnalyzerPerformanceInfo(AnalyzerInfoCache),
                             // +1 for project itself
-                            project.DocumentIds.Count + 1
-                        },
-                        callbackTarget: null,
-                        cancellationToken).ConfigureAwait(false);
-                }
-                catch (Exception ex) when (FatalError.ReportWithoutCrashUnlessCanceled(ex))
-                {
-                    // ignore all, this is fire and forget method
-                }
+                            documentAnalysisScope != null ? 1 : project.DocumentIds.Count + 1
```

Comment:
>documentAnalysisScope != null ? 1 : project.DocumentIds.Count + 1 [](start = 28, length = 65)

Store the value in a local, so that we know what this argument is by looking at the name of the local.